src/courses/models.py

```diff
@@ -470,6 +474,7 @@ def calc_semester_grades(self, semester):
         coursestudents = self.get_queryset().get_semester(semester)
         for coursestudent in coursestudents:
             coursestudent.final_xp = coursestudent.user.profile.xp_per_course()
+            assert coursestudent.final_xp >= 0, f"{coursestudent.user.get_full_name()} has a negative XP. Fix it before closing the semester"
```

Comment:
How is this going to inform the user?

Reply:
The method complete_active_semester() catches this exception and return Semester.STUDENTS_WITH_NEGATIVE_XP which is in turn handled by the view.

Reply:
Ok, let's remove the assert, we don't need that in our production code.